src/copyAs.ts

```diff
@@ -0,0 +1,118 @@
+import * as base64 from "js-base64";
+import * as vscode from "vscode";
+import { CopyFormat, ExtensionHostMessageHandler, MessageType } from "../shared/protocol";
+
+interface QuickPickCopyFormat extends vscode.QuickPickItem {
+	label: CopyFormat;
+}
+
+export const copyAs = async (messaging: ExtensionHostMessageHandler): Promise<void> => {
+	const formats: QuickPickCopyFormat[] = [
+		{ label: CopyFormat.Hex },
+		{ label: CopyFormat.Literal },
+		{ label: CopyFormat.Utf8 },
+		{ label: CopyFormat.C },
+		{ label: CopyFormat.Go },
+		{ label: CopyFormat.Java },
+		{ label: CopyFormat.JSON },
+		{ label: CopyFormat.Base64 },
+	];
+
+	vscode.window.showQuickPick(formats).then(format => {
+		if (format) {
+			messaging.sendEvent({ type: MessageType.TriggerCopyAs, format: format["label"] });
+		}
+	});
+};
+
+export function copyAsText(buffer: Uint8Array) {
+	vscode.env.clipboard.writeText(new TextDecoder().decode(buffer));
+}
+
+export function copyAsHex(buffer: Uint8Array) {
+	vscode.env.clipboard.writeText(Buffer.from(buffer).toString("hex"));
+}
+
+export function copyAsLiteral(buffer: Uint8Array) {
+	let encoded: string = "";
+	const digits = Buffer.from(buffer)
+		.toString("hex")
```

Comment:
didn't catch this before, but the hex editor runs in the web (e.g. vscode.dev) and these functions should work there too where Buffer isn't available. Fortunately this is easy to do with something like `Array.from(buffer, b => b.toString(16).padStart(2, '0')).join('')`

Reply:
I realized there is another problem with the web version, the 'process' variable results undefined. I added a check in the second to last commit